src/framework/ui/internal/platform/windows/windowsplatformtheme.cpp

```diff
@@ -0,0 +1,139 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "windowsplatformtheme.h"
+#include "log.h"
+#include <Windows.h>
+
+using namespace mu::ui;
+using namespace mu::async;
+
+using ThemeType = IUiConfiguration::ThemeType;
+
+static const std::wstring windowsThemeKey = L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
+static const std::wstring windowsThemeSubkey = L"AppsUseLightTheme";
+
+HKEY hKey = nullptr;
+
+WindowsPlatformTheme::WindowsPlatformTheme()
+{
+    using fnRtlGetNtVersionNumbers = void(WINAPI*)(LPDWORD major, LPDWORD minor, LPDWORD build);
+    auto rtlGetNtVersionNumbers
+        = reinterpret_cast<fnRtlGetNtVersionNumbers>(GetProcAddress(GetModuleHandleW(L"ntdll.dll"), "RtlGetNtVersionNumbers"));
+    if (rtlGetNtVersionNumbers) {
+        DWORD major, minor, buildNumber;
+        rtlGetNtVersionNumbers(&major, &minor, &buildNumber);
+        m_buildNumber = buildNumber & ~0xF0000000;
+    }
+}
+
+WindowsPlatformTheme::~WindowsPlatformTheme()
+{
+    stopListening();
+}
+
+void WindowsPlatformTheme::init()
+{
+    updateListeningStatus(configuration()->preferredThemeType());
+
+    configuration()->preferredThemeTypeChanged().onReceive(nullptr, [this](const ThemeType& newThemeType) {
```

Comment:
better add in the `windowsplatformtheme.h `

```
#include "async/asyncable.h"

class WindowsPlatformTheme : public IPlatformTheme, public async::Asyncable 
```

and here 

```
configuration()->preferredThemeTypeChanged().onReceive(this, [this](const ThemeType& newThemeType) {
```

it is an automatic disconnect mechanism